Language.java

```diff
@@ -0,0 +1,179 @@
+		Map<Character, Set<Character>> graph = new HashMap<Character, Set<Character>>();
+		Map<Character, Integer> degree = new HashMap<Character, Integer>();
+		Set<Character> uniqueChars = uniqueCharacters();
+		
+		if(sortedWords.size()==1){
+			String currentWord = sortedWords.get(0);
+			for(char c: currentWord.toCharArray()) {
+				graph.putIfAbsent(c, new HashSet<>());
+				degree.putIfAbsent(c, 0);
+			}
+		} else {
+			for (int i = 0; i < sortedWords.size() - 1; i++) {
+				String currenttWord = sortedWords.get(i);
+				String adjacentWord = sortedWords.get(i + 1);
+	
+				boolean wait = true;
+				int count = 0;
+				int limit = Math.min(currenttWord.length(), adjacentWord.length());
+				//Don't enter if word is empty
+				wait = wait && count < limit;
+				
+				while (wait) {
+					if (currenttWord.charAt(count) != adjacentWord.charAt((count))) {
+						Set<Character> set = graph.get(currenttWord.charAt(count));
+						if (set == null) {
+							set = new HashSet<>();
+							graph.put(currenttWord.charAt(count), set);
+						}
+						set.add(adjacentWord.charAt(count));
+						uniqueChars.remove(currenttWord.charAt(count));
+						// Make sure that vertices with zero indegree will be in the
+						// degree map
+						degree.putIfAbsent(currenttWord.charAt(count), 0);
+	
+						// Keep track of the indegree of every vertex while
+						// constructing the graph
+						int deg = degree.getOrDefault(adjacentWord.charAt(count), 0);
+						degree.put(adjacentWord.charAt(count), deg + 1);
+	
+						wait = false;
+					}
+					count++;
+					wait = wait && count < limit;
+				}
+			}
+		}
+		
+		//Make sure that all letters appear  in graph
+		for(char c: uniqueChars) {
+			graph.putIfAbsent(c, new HashSet<>());
+			degree.putIfAbsent(c, 0);
+		}
+		
+		return new Pair<Map<Character, Set<Character>>, Map<Character, Integer>>(graph, degree);
+	}
+	
+	/**
+	 * Returns a set of unique characters
+	 */
+	private Set<Character> uniqueCharacters() {
+		Set<Character> characters = new HashSet<>();
+		
+		for(String word: sortedWords) {
+			for(char c: word.toCharArray()){
+				characters.add(c); 
+			}
+		}
+		
+		return characters; 
+	}
+
+	/**
+	 * @param degree - Indegrees of the vertices from a given graph
+	 * @return A set of vertices with zero indegree
+	 */
+	private Set<Character> findStartVertices(Map<Character, Integer> degree) {
+		Set<Character> vertices = new HashSet<>();
+
+		for (Character character : degree.keySet()) {
+			if (degree.get(character) == 0) {
+				vertices.add(character);
+			}
+		}
+		return vertices;
+	}
+
+	/**
+	 * Topological sort of the vertices of an acyclic graph
+	 * 
+	 * For the language with sorted words [ART, RAT, CAT, CAR] 
+	 * valid sorts are: 
+	 * [A, T, R, C] and [T, A, R, C]
+	 * 
+	 * return - A list of sorted vertices.  
+	 * 
+	 */
+	private List<Character> topologicalSort(Map<Character, Set<Character>> graph, Map<Character, Integer> degree) {
```

Comment:
I wasn't familiar with this implementation of the topological sort, I only knew the DFS based one, this is good too.